src/array_string.rs

```diff
@@ -565,3 +573,19 @@ impl<'de, A> Deserialize<'de> for ArrayString<A>
         deserializer.deserialize_str(ArrayStringVisitor::<A>(PhantomData))
     }
 }
+
+#[cfg(feature="diesel")]
+/// Requires crate feature `"diesel"`
+impl<A, ST, DB> FromSql<ST, DB> for ArrayString<A>
+    where A: Array<Item = u8> + Copy,
+          DB: Backend,
+          *const str: FromSql<ST, DB>
+{
+    fn from_sql(bytes: Option<&DB::RawValue>) -> diesel::deserialize::Result<Self> {
+        let str_ptr = <*const str as FromSql<ST, DB>>::from_sql(bytes)?;
```

Comment:
What's the argument for this code being correct? What does FromSql do for `*const str` ?